### PR TITLE
Dev deps fix + cleanup + refactor

### LINKED ISSUE #90: Don't copy devDependencies
there is an issue about this problem but it's closed.
https://github.com/graphcool/serverless-plugin-typescript/issues/13

Some other ppl and I still have the problem in that issue. So just to reopen it here.

package.json
```
{
  "name": "aws-credentials-exp",
  "version": "1.0.0",
  "description": "",
  "main": "handler.js",
  "dependencies": {
    "aws-sdk": "^2.201.0"
  },
  "devDependencies": {
    "serverless-plugin-typescript": "^1.1.5"
  },
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}
```

serverless-plugin-typescript is not in the final release, but typescript and the dependencies of typescript do.

I tried not to include serverless-plugin-typescript plugin in serverless.yml (it's still installed and exist in package.json), the node_modules in zip file will only have aws-sdk and its dependencies.

### PR DESCRIPTION
**Pull request by JackCuthbert**
Sorry for the delay in getting around to some of these fixes, it's been hectic at work lately so I hope this PR makes up for it :slightly_smiling_face: 

* I've added a declaration file (and a small change to tslint to allow this) to make type definitions a little easier to use during development
* I've included the development dependencies fix which is based on the work in https://github.com/prisma/serverless-plugin-typescript/pull/157 by @MennoTammens and @jengjeng (thanks!). This should resolve #90 and #154.
* I've refactored the hooks property to better illustrate how the build works now that we're passing in a flag to determine when to package/exclude dependencies
* I've added `typescript` to the installation instructions in the readme (https://github.com/prisma/serverless-plugin-typescript/issues/151)
* There's also a general cleanup of unused locals, formatting, and style

For those that want to review this (hey @divyenduz :stuck_out_tongue:), the commits should be relatively atomic making it easier if you go through one commit at a time.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
     "@types/fs-extra": "5.0.5",
     "@types/jest": "24.0.12",
     "@types/lodash": "4.14.123",
+    "@types/node": "^12.6.2",
```
Comment (divyenduz) [inline package.json]: Doest't block this PR but in this or or separately, we can pin deps like this and `rimraf`. Maybe use green keeper or something (in future) to always pin deps. I miss this all the time myself as the defaults (override-able via `.npmrc` I think) don't pin versions by default.

**Comment by JackCuthbert** [comment]
> btw is the semantic release still working?

Hmm, doesn't appear to be since the last merge, I think I omitted the semantic-release format on accident :fearful: This one should trigger it

File: package.json (hunk 1 above)
Comment (JackCuthbert) [reply package.json]: Good catch, updated :+1:

**Comment by divyenduz** [comment]
I will patch it up 👍

**Pull request merged**